### PR TITLE
Fix /transaction/pool with nonce-gaps

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- when calling /transaction/pool with nonce-gaps, it does not check sender's account nonce, thus leading to missing the first gap that led to accumulating txs in pool
  
## Proposed changes
- check sender's account nonce as well
- apply ToLower on fields

## Testing procedure
- system test
- scenario:
   - generate some txs with different nonces from a specific sender, but make sure to break the nonces so they go to pool
   - call `/address/erd1.../nonce`, where erd1... is the sender address used in the above step
   - **on a node that is in the same shard with the sender**(or on a proxy, but make sure the proxy is started with `AllowEntireTxPoolFetch = true` in config.toml) call `/transaction/pool?by-sender=erd1...&nonce-gaps=true` to get the nonce gaps for this account
   - make sure that the gap starting from the sender's account nonce to the first nonce in pool is also returned by the transactions pool api
   - make sure following calls work also:
      - `/transaction/pool?by-sender=erd1...&fields=gasprice`
      -  `/transaction/pool?by-sender=erd1...&fields=GASPRICE`
      -  `/transaction/pool?by-sender=erd1...&fields=gasPrice`

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by SebastianMarian — APPROVED** [review]

File: node/external/transactionAPI/apiTransactionProcessor.go [hunk 1]
```diff
@@ -378,13 +378,23 @@ func (atp *apiTransactionProcessor) fetchLastNonceForSender(sender string, sende
...
 
-func (atp *apiTransactionProcessor) extractNonceGaps(sender string, senderShard uint32) ([]common.NonceGapApiResponse, error) {
+func (atp *apiTransactionProcessor) extractNonceGaps(sender string, senderShard uint32, senderAccountNonce uint64) ([]common.NonceGapApiResponse, error) {
 	wrappedTxs := atp.fetchTxsForSender(sender, senderShard)
 	if len(wrappedTxs) == 0 {
 		return []common.NonceGapApiResponse{}, nil
 	}
 
 	nonceGaps := make([]common.NonceGapApiResponse, 0)
+	firstNonceInPool := wrappedTxs[0].Tx.GetNonce()
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply node/external/transactionAPI/apiTransactionProcessor.go:388]: pushed

File: node/external/transactionAPI/apiTransactionProcessor.go [hunk 2]
```diff
@@ -411,6 +404,26 @@ func (atp *apiTransactionProcessor) extractNonceGaps(sender string, senderShard
...
+	firstNonceInPool uint64,
+	senderShard uint32,
+	nonceGaps *[]common.NonceGapApiResponse,
+) {
+	if atp.shardCoordinator.SelfId() != senderShard {
+		return
+	}
+
+	nonceDif := firstNonceInPool - senderAccountNonce
+	if nonceDif > 1 {
```
Comment (bogdan-rosianu) [inline node/external/transactionAPI/apiTransactionProcessor.go:418]: This doesn't cover all the situations. For example, considering the wallet of Alice with nonce = 5. In the pool, the smallest Alice's nonce is 6. Therefore, `6-5=1`, and `1` is not greater than `1`, so nothing will be appended. 

So it won't work for a single nonce gap.

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: node/external/transactionAPI/apiTransactionProcessor.go [hunk 3]
```diff
@@ -411,6 +404,26 @@ func (atp *apiTransactionProcessor) extractNonceGaps(sender string, senderShard
...
+		nonceGap := common.NonceGapApiResponse{
+			From: senderAccountNonce + 1,
```
Comment (SebastianMarian) [inline node/external/transactionAPI/apiTransactionProcessor.go:420]: From: senderAccountNonce,

Comment (sstanculeanu) [reply node/external/transactionAPI/apiTransactionProcessor.go]: pushed

File: node/external/transactionAPI/apiTransactionProcessor.go (hunk 2 above)
Comment (sstanculeanu) [reply node/external/transactionAPI/apiTransactionProcessor.go]: pushed

**Review by bogdan-rosianu — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Pull request merged**